### ISSUE TITLE
Add an Item - Tags - 'Remove Tag' buttons lack a descriptive name and are announced as links

### ISSUE THREAD

**Issue by aviannamiller** [omeka/Omeka#1128]
"Description: On the 'Add an Item' page is a 'Tags' tab. On this page, users are able to add tags to the item by entering text into the 'Add Tags' edit field and then selecting the 'Add Tags' button. These tags appear on the screen after they've been added below an 'All Tags' heading. However, when these tags receive focus, they are currently announced as 'Remove, Same Page Link'. This can be confusing for users in two ways:

1) The accessible name of 'Remove' is not descriptive and does not tell the user what they are removing. The associated text of the tag is not announced nor is it clear if the user is removing a 'Tag'.
2) The 'Remove' buttons are currently announced as a link erroneously. A link implies a navigation action will occur, whilest a button is an action on the same page. Toggling these 'Links' changes the state of the button from green to red and the button changes it's name to 'Undo, Same Page Link'. 

Expected Results: All interactive components have an accessible name, a clear role, and the current state/value as needed. The 'Remove'/'Undo' button would announce as a button and have an accessible name similar to '{tag name}, Remove Tag' button. 
Note:
Links are navigational in purpose such as going to another internal page on the site.
Buttons tend to be actions on the same page such as opening a modal or submitting a form.

Actual Results: The 'Remove' button has a non-descriptive name as well as is mistakenly announced as a link."

4.1.2 Name, Role, Value; Level A; WCAG 2.0

**Comment by kimisgold** [comment]
A fix is ready for testing on the [`a11y/tag-actions`](https://github.com/omeka/Omeka/tree/a11y/tag-actions) branch. Tags should now have labeled group containers to make content relationships clearer, and the anchors have been converted to buttons.

**Comment by aviannamiller** [comment]
Something's not working with the button styling when you add a new tag.

<img width="473" height="119" alt="Tags section on item edit Tags tab. Tags are present, labelled: 'tag', 'test', 'the', 'new tag'. The last two tags in the group have styling issues with their remove / undo butttons." src="https://github.com/user-attachments/assets/7de95ce1-8cc4-4974-be7d-0b0d4642deed" />

The newly added tags also do not save.

**Comment by aviannamiller** [comment]
Otherwise, the adjusted labelling is working well. You do get some repetition of tag name with JAWS/Firefox, but other than that, things sound good. Is it possible to also include the tag count with the label on the browse view?
